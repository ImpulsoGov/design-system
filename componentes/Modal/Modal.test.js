import { Modal } from ".";
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";

const COMPONENT = "Modal";
const children = <p>Teste</p>;
const scenarios = [
  {
    show: true,
    setModal: jest.fn(),
  },
  {
    setModal: jest.fn(),
  }
];

describe(`Componente: ${COMPONENT}`, () => {
  it('deve renderizar corretamente', () => {
    render(<Modal {...scenarios[0]}>{children}</Modal>);

    const component = screen.getByTestId("Modal");

    expect(component).toMatchSnapshot();
  });

  it('deve chamar setModal com o valor false ao clicar no botão de fechar', async () => {
    const user = userEvent.setup();
    render(<Modal {...scenarios[0]}>{children}</Modal>);

    const closeBtn = screen.getByTestId("ModalExit");
    await user.click(closeBtn);

    expect(scenarios[0].setModal).toHaveBeenCalledWith(false);
  });

  it('não deve ser renderizado quando a visibilidade inicial é false', () => {
    render(<Modal {...scenarios[1]}>{children}</Modal>);

    const component = screen.queryByTestId("Modal");

    expect(component).not.toBeInTheDocument();
  });
});