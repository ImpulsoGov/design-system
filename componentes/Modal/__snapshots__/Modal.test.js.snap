// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Componente: Modal deve renderizar corretamente 1`] = `
<div
  class="ModalContainer"
  data-testid="Modal"
>
  <div
    class="ModalExit"
    data-testid="ModalExit"
  />
  <p>
    Teste
  </p>
</div>
`;
